istioctl/cmd/proxyconfig.go

```diff
@@ -860,19 +863,19 @@ func bootstrapConfigCmd() *cobra.Command {
 			if err != nil {
 				return err
 			}
+
 			switch outputFormat {
-			case yamlOutput:
+			case summaryOutput:
+				return configWriter.PrintVersionSummary()
+			case jsonOutput, yamlOutput:
 				return configWriter.PrintBootstrapDump(outputFormat)
 			default:
-				return configWriter.PrintBootstrapDump(jsonOutput)
-				// TODO: cobra default value of global flag is passed instead of local flag
-				// i.e short is passed instead of json as declared below
-				// return fmt.Errorf("output format %q not supported", outputFormat)
+				return fmt.Errorf("output format %q not supported", outputFormat)
 			}
 		},
 	}
 
-	bootstrapConfigCmd.Flags().StringVarP(&outputFormat, "output", "o", jsonOutput, "Output format: one of json|yaml")
+	bootstrapConfigCmd.Flags().StringVarP(&outputFormat, "output", "o", jsonOutput, "Output format: one of json|yaml|short")
```

Comment:
@howardjohn @shamsher31 

The default value here is being overridden by a different command's flag setup since the commands all share a global and use `-o short` by default, which is changing the default behavior to be the new version summary versus JSON causing the istioctl pilot tests to fail.

Do we want `-o short` to be default behavior across the board for the entire command tree, if so, I'll update the tests to explicitly add a `-o json` flag. Otherwise, I'll just shadow the `outputFormat` variable in the function so it respects the json default.

Reply:
my 2 cents - probably not good to change the default behavior on this, which I missed earlier while reviewing it.   There likely users building automations on this today and could break them if we change `-o short` to be the default.

Reply:
I tend to agree, the only real issue I see is that output is global to the proxy-config command so we'd be overwriting the default value based on the subcommand. Probably fine, just something worth noting.